### PR TITLE
Galactic support for the Dockerfile

### PR DESCRIPTION
**Pull request by j-rivero**
The PR implements the Galactic support for the current Dockerfile. It was not difficult and the only big problem found were the compiler errors of default clang in Ubuntu Focal. Steps:

 * Parametrize the Dockerfile to support build-args UBUNTU_VERSION and ROS_VERSION 985d23a0bccd4c45665bf48894dec7cb9fb430aa
 * Use clang-12 and on Focal install CMake 3.14 (thanks to @spham-amzn) 6bde42a3b1a02ca5324664f1aafa26e751f07606
 * Update README instructions deb535838ac12aad929c4b72479f68b7099e7d6a

I've changed the README to just use the raw Dockerfile instead of cloning the whole repository.  Added the Galactic instructions and  update the repository URLs.

Fix: #196

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
-the demo so it can take several hours depending on the Internet connection.
+## Building the Docker Image using Ubuntu Focal and ROS 2 Galactic
+
+Creating the image using the Dockerfile to have an Ubuntu Focal and ROS 2
+Galactic platform requires just to have docker installed and provide some
+extra building arguments to it:
+
+```
+wget https://raw.githubusercontent.com/o3de/ROSConDemo/main/docker/Dockerfile
+docker build -t roscon_demo_galactic --build-arg ROS_VERSION=galactic --build-arg UBUNTU_VERSION=focal  .
```
Comment (michalpelka) [inline docker/README.md:38]: I've tried to build this image. It failed on updating CMake.
Here is log for you: https://gist.github.com/michalpelka/59f80dff9c9a98aa50935f31ec3a897e

**Review by michalpelka — COMMENTED** [review]
There is a problem with Focal/Galaxy.

**Comment by adamdbrw** [comment]
@j-rivero could we rebase and merge this one?

**Comment by j-rivero** [comment]
> @j-rivero could we rebase and merge this one?

Rebase seems a bit hard due to the number of conflicts with the `main` branch. Does a merge commit or an squash works for you?

**Comment by michalpelka** [comment]
@j-rivero I think that we can squash and merge it.
I was able to build it in galactic and humble. Although I have an issue with rviz2 with focal/galactic:
```
[ERROR] [1669800056.668294920] [rviz2]: rviz::RenderSystem: error creating render window: InvalidParametersException: Window with name 'OgreWindow(0)' already exists in GLRenderSystem::_createRenderWindow at /tmp/binarydeb/ros-galactic-rviz-ogre-vendor-8.5.1/.obj-x86_64-linux-gnu/ogre-v1.12.1-prefix/src/ogre-v1.12.1/RenderSystems/GL/src/OgreGLRenderSystem.cpp (line 1061)
[ERROR] [1669800056.668299909] [rviz2]: InvalidParametersException: Window with name 'OgreWindow(0)' already exists in GLRenderSystem::_createRenderWindow at /tmp/binarydeb/ros-galactic-rviz-ogre-vendor-8.5.1/.obj-x86_64-linux-gnu/ogre-v1.12.1-prefix/src/ogre-v1.12.1/RenderSystems/GL/src/OgreGLRenderSystem.cpp (line 1061)
```
It seems that is an existing problem with rviz: https://github.com/ros2/rviz/issues/753

**Comment by j-rivero** [comment]
> Although I have an issue with rviz2 with focal/galactic:

Let me try to reproduce it

**Comment by michalpelka** [comment]
We agreed that we can merge it - the issue with rviz on galactic is not crucial.

**Pull request merged**